### PR TITLE
Added support for 128 bit TM/HM learnsets

### PR DESCRIPTION
**Pull request by StephenLynx**
First of all, the gTMHMLearnsets was changed to an array of 4 32 bit numbers. 
Then the macro was changed to take in account tms that belong to the second half.
Consequently the function to check if a mon can learn a tm move was adapted.
I also added the defines necessary to both the list of tm items that take the name of the move as the list that associates tms with moves.

edit: I wasn't able to test it with actual tms, but it runs.

Stephen Lynx#9875

### CODE REVIEW

File: src/pokemon.c [hunk 1]
```diff
@@ -7302,39 +7302,27 @@ bool8 TryIncrementMonLevel(struct Pokemon *mon)
...
-    {
-        u32 mask = 1 << (tm - 32);
-        return gTMHMLearnsets[species][1] & mask;
-    }
+
+    //First we obtain which of the 4 32 bit integers the tm falls into
+    //The alternative would be an if-else for each range
+    index = (tm - (tm % 32)) / 32;
+
+    //We shift the bits based on the index
```
Comment (LOuroboros) [inline src/pokemon.c:7322]:
```suggestion
    // First we obtain which of the 4 32 bit integers the tm falls into
    // The alternative would be an if-else for each range
    index = (tm - (tm % 32)) / 32;

    // We shift the bits based on the index
```

**Review by LOuroboros — COMMENTED** [review]
I'm not exactly a fan of how the transition from the 2nd to the 3rd array(?) looks personally.

```
                                             | TMHM(TM45_ATTRACT)
                                            , TMHM(HM01_CUT)
                                             | TMHM(HM04_STRENGTH)
```

If this can't be handled with a single macro in a way that looks natural, I think it'd be better to add a new one for the TM64-TM128 arrays like `TMHM2`? Something that signals to the user that a new pair of arrays is being used to handle that part of a TM/HM learnset. If I'm not misremembering, that's also what other people who applies this sort of thing to their project do.

By the way, you deleted the TM/HM learnsets for Gen. 4+ species which is not good.

**Comment by StephenLynx** [comment]
I can just adapt the script that generates the final thing. I didn't notice there were other learnsets, I'll change that.

**Comment by AsparagusEduardo** [comment]
For future reference, you can change the target branch in github web.

**Pull request closed without merging**